.github/workflows/ci-ipfs.yml

```diff
@@ -0,0 +1,82 @@
+name: Deploy IPFS Base
+
+on:
+  workflow_call:
+    inputs:
+      ipfs_folder:
+        description: 'IPFS build folder'
+        default: 'out/'
+        type: string
+      W3S_TOKEN:
+        description: 'web3storage token'
+        type: string
+      GW3_TOKEN:
+        description: 'gateway3 token'
+        type: string
+
+permissions:
+  contents: write
+
+jobs:
+  deploy-ipfs:
+    runs-on: ubuntu-latest
+    name: Build and deploy
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Install node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16
+          cache: 'yarn'
+      - name: Extract branch name
+        id: extract_branch
+        run: |
+          echo "BRANCH_NAME=${GITHUB_HEAD_REF:-${GITHUB_REF#refs/heads/}}" >> $GITHUB_OUTPUT
+      - name: Install dependencies
+        run: |
+          yarn install --frozen-lockfile
+      - name: Build
+        run: |
+          yarn build-ipfs
+      - name: IPFS pinning
+        id: pinning
+        run: |
+          export BODY_RESULT=`npx -y blumen@0.0.0 deploy $IPFS_FOLDER`
+          echo "$BODY_RESULT"
+          echo "BODY_RESULT<<EOF" >> $GITHUB_OUTPUT   
+          echo "$BODY_RESULT" >> $GITHUB_OUTPUT   
+          echo "EOF" >> $GITHUB_OUTPUT
+        env:
+          BLUMEN_W3S_TOKEN: ${{ inputs.W3S_TOKEN }}
+          BLUMEN_GW3_TOKEN: ${{ inputs.GW3_TOKEN }}
+          IPFS_FOLDER: ${{ inputs.ipfs_folder }}
+      - name: Zip
+        if: steps.extract_branch.outputs.BRANCH_NAME == 'main'
+        id: zip
+        run: |
+          zip -r $FILE_NAME $IPFS_FOLDER
+          echo "FILE_NAME=$FILE_NAME" >> $GITHUB_OUTPUT
+        env:
+          FILE_NAME: ipfs_source_code.zip
```

Comment:
if it's a constant, you can assign it per workflow to not use steps outputs

Reply:
Could you show an example?

Reply:
https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#env just env on the higher level